src/NUnitTestAdapter/NUnit3TestDiscoverer.cs

```diff
@@ -97,6 +91,11 @@ public void DiscoverTests(IEnumerable<string> sources, IDiscoveryContext discove
                         int cases = ProcessTestCases(topNode, discoverySink, testConverter);
 
                         TestLog.Debug(string.Format("Discovered {0} test cases", cases));
+                        // Only save if seed is not specified in runsettings
+                        // This allows workaround in case there is no valid
+                        // location in which the seed may be saved.
+                        if (cases>0 && !Settings.RandomSeedSpecified)
+                            Settings.SaveRandomSeed(Path.GetDirectoryName(sourceAssemblyPath));
```

Comment:
The code was copied over, no changes otherwise.   It will generate different values for each run, but that's the same as it always has been.  
There is a test for the random seed, and for the reading of that from the runconfig, but not any for running with these.   It would be good to wrap the discoverer and the executor with tests around them, but that would require mocking of the connections or using some fake assemblies.  The older tests we had for the AssemblyRunner are commented out.  There is a TODO there to convert "as needed".    This could be improved, but I think that is beside this fix, although it would have benefited from it.